node/src/reactor/validator.rs

```diff
@@ -1038,9 +1038,14 @@ impl reactor::Reactor for Reactor {
                     Box::new(block.header().clone()),
                 ));
                 let reactor_event_es =
-                    Event::EventStreamServer(event_stream_server::Event::BlockAdded(block));
+                    Event::EventStreamServer(event_stream_server::Event::BlockAdded(block.clone()));
                 let mut effects = self.dispatch_event(effect_builder, rng, reactor_event_es);
                 effects.extend(self.dispatch_event(effect_builder, rng, reactor_event_consensus));
+                effects.extend(self.dispatch_event(
+                    effect_builder,
+                    rng,
+                    small_network::Event::from(LinearChainAnnouncement::BlockAdded(block)).into(),
```

Comment:
Can you construct the `small_network::Event` manually? Those implicit mappings have already been a problem in the past. We seem to construct events manually anyway (at least in the couple of lines above this one).

Reply:
The issue I have with the manual construction is that it moves the knowledge of how to construct the small network's effect variant from the `LinearChainAnnouncement` into the router, instead of keeping it in the `event` submodule. Even "inspecting" these events by matching on variants is not using the system as intended (by me, at least :))

This is, of course, not the PR to clean this up though, so I can either leave it be or change it to be in line with the existing code though, whichever you prefer :)